storage/google/cloud/storage/notification.py

```diff
@@ -82,6 +90,40 @@ def __init__(self, bucket, topic_name,
         if payload_format is not None:
             self._properties['payload_format'] = payload_format
 
+    @classmethod
+    def from_api_repr(cls, resource, bucket):
+        """Construct an instance from the JSON repr returned by the server.
+
+        See: https://cloud.google.com/storage/docs/json_api/v1/notifications
+
+        :type resource: dict
+        :param resource: JSON repr of the notification
+
+        :type bucket: :class:`google.cloud.storage.bucket.Bucket`
+        :param bucket: Bucket to which the notification is bound.
+
+        :rtype: :class:`BucketNotification`
+        :returns: the new notification instance
+        :raises ValueError:
+            if resource is missing 'topic' key, or if it is not formatted
+            per documented spec.
+        """
+        topic_path = resource.get('topic')
+        if topic_path is None:
+            raise ValueError('Resource has no topic')
+
+        match = _TOPIC_REF_RE.match(topic_path)
+        if match is None:
+            raise ValueError(
+                'Resource has invalid topic: {}'.format(topic_path))
```

Comment:
You could include the regex in this error message? (I assume it has a not-so-bad repr?)

Reply:
998b9f9 puts the URL of the spec in the message.